openpype/hosts/maya/plugins/publish/extract_look.py

```diff
@@ -58,6 +59,11 @@ def maketx(source, destination, *args):
     from openpype.lib import get_oiio_tools_path
 
     maketx_path = get_oiio_tools_path("maketx")
+
+    if platform.system().lower() == "windows":
+        # Ensure .exe extension
+        maketx_path += ".exe"
+
```

Comment:
This isn't necessarily due to Python 3, but without this it would fail on Windows because the executable is actually `maketx.exe`.

Reply:
Would it be better if we change the following line
```python
if not os.path.exists(maketx_path):
    ...
```
into
```python
if not (os.path.exists(maketx_path) or os.path.exists(maketx_path + ".exe")):
    ...
```
instead ?

Thought this would be much clearer about what it is for. (I firstly thought it's for the subprocess call)

Reply:
Maybe yes. However if the `.exe` were to exist on a Mac or Linux system it still wouldn't be valid to run `maketx` so having it more explicit might make more sense?

Reply:
Hmmm, in that case, how about also changes the following lines `if os.path.exists()...` into
```python
...
if not os.access(maketx_path, os.X_OK):
    raise AssertionError("Not an executable: %s" % maketx_path)
```

Reply:
If that works without the `.exe` and is cross-OS compatible that sounds like the best bet but it's hard for me to test in production. @antirotor (I noticed you're set as reviewer), what are your thoughts?

Reply:
> If that works without the `.exe `

Ohoh I didn't mean to use `os.access` to replace the `.exe` appending change, but do both :)

However the `os.access` does require appending `.exe` explicitly.

Reply:
> However the `os.access` does require appending `.exe` explicitly.

In that case still `avalon.lib.which` might be the best approach - since that also does that same `os.access` check. However... the only difference is that would allow falling back to any other `maketx` on a user's `PATH` which might not be as explicit as we'd want if we want to make 100% sure the OpenPype `maketx` is used.

Reply:
problem with `os.access()` is that it is not 100% reliable. We had cases where it didn't reported correctly `os.X_OK` probably because some storage configuration.

Reply:
@antirotor was there a specific way you'd like me to change/implement this?

Reply:
I think the original approach with platform check and appending `.exe` is the safest. Not that nice or sofisticated, but it doesn't rely on `os.access`. If it is executable but it is not executable :) than imho it is resposibility of admin to set it so. We provide `maketx` with openpype and I suppose the only way when maketx is present but is not executable is some kind of wrong build anyway.

Reply:
Thanks. Then for now I'll resolve this conversation.